cli/user/src/test/java/io/pravega/cli/user/config/InteractiveConfigCommandTest.java

```diff
@@ -54,4 +57,28 @@ public void testSetConfig() {
         interactiveConfig.set(InteractiveConfig.LOG_LEVEL, "debug");
         Assert.assertEquals(interactiveConfig.getLogLevel(), Level.DEBUG);
     }
+
+    @Test
+    public void testGetDefaultsForTlsEnv() throws NoSuchFieldException, IllegalAccessException {
+        Map<String, String> unmodifiableEnv = System.getenv();
+        Field field = unmodifiableEnv.getClass().getDeclaredField("m");
+        field.setAccessible(true);
+        Map<String, String> modifiableEnv = (Map<String, String>) field.get(unmodifiableEnv);
+        modifiableEnv.put("PRAVEGA_CONTROLLER_URI", "tls://testControllerURI");
+        InteractiveConfig interactiveConfig = InteractiveConfig.getDefault();
+        Assert.assertEquals("testControllerURI", interactiveConfig.getControllerUri());
+        Assert.assertEquals(true, interactiveConfig.isTlsEnabled());
+    }
+
+    @Test
+    public void testGetDefaultsForNonTlsEnv() throws NoSuchFieldException, IllegalAccessException {
+        Map<String, String> unmodifiableEnv = System.getenv();
+        Field field = unmodifiableEnv.getClass().getDeclaredField("m");
```

Comment:
I was trying to understand why you had used reflection to modify the `System.Env` but then I realised it's because the `InteractiveConfig::getDefault` isn't test-friendly and assumes the System Environment is available.

Instead, we could change `InteractiveConfig` to a more test-friendly version:
```
    public static InteractiveConfig getDefault(Map<String, String> env) {
        //Default tls based configurations for pravega cli
        boolean tlsEnabled = false;
        String systemEnvURI = env.get("PRAVEGA_CONTROLLER_URI");
        String controllerURI = systemEnvURI != null ? systemEnvURI : "localhost:9090";
        if (controllerURI.startsWith("tls://")) {
            tlsEnabled = true;
            controllerURI = controllerURI.replace("tls://", "");
        }
```
And your test is now much easier to understand:
```
    @Test
    public void testGetDefaultsForTlsEnv() throws NoSuchFieldException, IllegalAccessException {
        Map<String, String> environment = Collections.singletonMap("PRAVEGA_CONTROLLER_URI", "tls://testControllerURI");
        
        InteractiveConfig interactiveConfig = InteractiveConfig.getDefault(environment);
        Assert.assertEquals("testControllerURI", interactiveConfig.getControllerUri());
        Assert.assertEquals(true, interactiveConfig.isTlsEnabled());
    }
```
And `doMain` becomes:
```
    @VisibleForTesting
    public static void doMain(String[] args, InputStream interactiveStream) {
        System.out.println("Pravega User CLI Tool.");
        System.out.println("\tUsage instructions: https://github.com/pravega/pravega/wiki/Pravega-User-CLI\n");
        val config = InteractiveConfig.getDefault(System.getenv());
```

Is there a reason you did `System.getenv()` and then used reflection to modify the `Map`?  It would be simpler to create a new `Map` and pass it in, i.e. 
```
        Map<String, String> modifiableEnv = Collections.singletonMap("PRAVEGA_CONTROLLER_URI", "tls://testControllerURI");
```
I realise in the running code these values would come from the environment, but here in test code they can be provided via any `Map`.

Reply:
Agree, we should have reverted the approach of expecting a map as input to the method, this helped us to get rid of reflection. Thanks for suggesting @maddisondavid .